denops/ddu-devicon/main.ts

```diff
@@ -34,37 +34,31 @@ export function getIconData(
   };
 }
 
-export function getSpecificFileIconData(
-  path: string,
-  specificFileIcons: Record<string, IconData>,
-  isFolder?: boolean,
-): IconData | undefined {
-  if (isFolder) {
-    return undefined;
-  }
-  const filename = basename(path);
-  return specificFileIcons[filename] ?? undefined;
-}
-
 export function getCustomIconData(
-  path: string,
-  customIconData: Record<string, IconData>,
+  filename: string,
+  specificFileIcon: Record<string, IconData>,
+  extentionIcon: Record<string, IconData>,
   isFolder?: boolean,
 ): IconData | undefined {
   if (isFolder) {
     return undefined;
   }
-  const def = getDeviconDef(path, isFolder);
-  const ext = (extname(path) === "")
-    ? basename(path)
-    : extname(path).slice(1, extname(path).length);
-  if (!customIconData[ext]) {
-    return undefined;
+  const def = getDeviconDef(filename, isFolder);
+  let customIcon = specificFileIcon[filename];
+  if (customIcon) {
+    return {
+      icon: customIcon.icon ?? (def?.icon ?? ""),
+      hl_group: customIcon.hl_group ?? (def && getHighlightGroup(def)),
+    };
+  }
+  const ext = extname(filename).slice(1, extname(filename).length);
```

Comment:
In this case, the second argument of Array#slice can be omitted and is equivalent.
```typescript
  const ext = extname(filename).slice(1);
```
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice